step-capstone/src/components/AddEvent.js

```diff
@@ -1,60 +1,72 @@
 import React, { useEffect } from 'react'
 import { useState } from 'react'
 import {
-    TextField,
-    Grid,
-    Checkbox,
-    FormControlLabel,
+ TextField,
+ Grid,
+ Checkbox,
+ FormControlLabel,
 } from '@material-ui/core';
 import {
-    MuiPickersUtilsProvider,
-    DateTimePicker
+ MuiPickersUtilsProvider,
+ DateTimePicker
 } from '@material-ui/pickers';
 import DateFnsUtils from "@date-io/date-fns";
-
-export default function AddEventHotel(props) {
+export default function AddEvent(props) {
     let overwriting = props.data !== undefined;
-
     // Sets values to previous values if editing, otherwise blank slate
-    const [startDate, handleStartChange] = useState(overwriting ? props.data.startDate : new Date());
-    const [endDate, handleEndChange] = useState(overwriting ? props.data.endDate : new Date());
+    const [startDate, setStartDate] = useState(overwriting ? props.data.startDate : props.startDate);
+    const [endDate, setEndDate] = useState(overwriting ? props.data.endDate : props.startDate);
     const [checked, setChecked] = useState(overwriting ? props.data.finalized : false);
     const [title, setTitle] = useState(overwriting ? props.data.title : "");
     const [location, setLocation] = useState(overwriting ? props.data.location : "");
     const [description, setDescription] = useState(overwriting ? props.data.description : "");
-
+    
     const handleChecked = (e) => {
         setChecked(e.target.checked);
     }
-
     const handleTitleChange = (e) => {
         setTitle(e.target.value);
     }
-
     const handleLocationChange = (e) => {
         setLocation(e.target.value);
     }
-
     const handleDescriptionChange = (e) => {
         setDescription(e.target.value);
     }
 
+    const handleStartDateChange = (newStartDate) => {
+        if (newStartDate >= endDate) {
+            var newEndDate = new Date(newStartDate);
+            newEndDate.setTime(newStartDate.getTime() + 5*60000);
+            setEndDate(newEndDate);
+        }
+        setStartDate(newStartDate);
+
+    }
+    const handleEndDateChange = (newEndDate) => {
+        if (startDate >= newEndDate) {
+            var newStartDate = new Date(newEndDate);
+            newStartDate.setTime(newEndDate.getTime() - 5*60000);
+            setStartDate(newStartDate);
+        }
+        setEndDate(newEndDate);
+    }
```

Comment:
These functions confuse me a little. It looks like their purpose is to update the start or end point to be a fixed unit of time from whichever point was just set. Could you clarify why these are here? Also, what are the time units on the offset you use?

Reply:
To clarify, these two functions ensure that `startDate` must be smaller than `endDate` (their names is Date but actually they are referring to Time; first I started with `timeStart` and `timeEnd`; but I refactored it to fit Emmie and Zach's code).
For example, If the user changes the time of the `endDate` to a time that is smaller than the current `startDate`, this function will change the `startDate` to be at least 5 minutes before the new `endDate`. Let me know if this explanation still confuses you.

Reply:
Instead of doing this implicit behavior, would it instead be possible to instead throw an error to the user to have them set the endDate to a time later than the startDate?

Reply:
In the current code, I try to replicate the behavior of Google Calendar, but yes, we can also throw the error to user. What do you think?

Reply:
From what I tried in Calendar, it seems that Calendar prevents you from creating an even that has endTime prior to startTime in the first place. I'm not sure which behavior you're referring to (maybe I'm misunderstanding) but I still prefer not allowing the user to create such an event in the first place (refuse to take input for event creation). Maybe @ganewman42 has a different opinion here.

Reply:
@cthamrun I think you have misunderstood my point. So the current behavior happens during the editting process. The user will has no chance to submit a TravelObject if that has invalidated time range, because the invalidated time range will be reset to be validated again. So the current behavior is: during the editing process, if the new endDate is smaller than the current startDate, the current startDate will be set to be 5 minutes before the new endDate; if the new startDate is bigger than the current endDate, the current endDate will be set to be 5 minutes after the new startDate.

Reply:
![Screenshot 2020-07-06 at 7 27 04 PM](https://user-images.githubusercontent.com/42331170/86666966-2645f880-bfbf-11ea-9974-017431448038.png)
For example, in Google Calendar, I have an event from 5:30 to 6

Reply:
If I change the start time to 6:15, it will automatically change the end time to be 6:45 
![Screenshot 2020-07-06 at 7 27 18 PM](https://user-images.githubusercontent.com/42331170/86667208-66a57680-bfbf-11ea-95a0-17afefac721a.png)